src/Query/Transformer/NamedArrayParameter.php

```diff
@@ -0,0 +1,54 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Linio\Component\Database\Query\Transformer;
+
+use Linio\Component\Database\Exception\InvalidQueryException;
+use Linio\Component\Database\Query\Transformer;
+
+class NamedArrayParameter implements Transformer
+{
+    public function execute(string &$query, array &$params = []): void
+    {
+        foreach ($params as $paramKey => $paramValue) {
+            $isNumeric = is_numeric($paramKey);
+            $isArray = is_array($paramValue);
+
+            if ($isNumeric && $isArray) {
+                throw new InvalidQueryException(sprintf('Unnamed parameter "%s" can\'t have array value.', $paramKey));
+            }
+
+            if ($isNumeric || !$isArray) {
+                continue;
+            }
+
+            $placeholders = $this->placeholders($paramKey, $paramValue);
+            $params += $placeholders;
```

Comment:
What I'm saying is that you should not change (add, unset) the `$params` array that is being looped over as it might create issues:

```php
foreach ($params as $paramKey => $paramValue) {
```

Instead, you should create a new array (e.g. `$finalParams`) that will contain the parameters that will be used.

```php
$finalParams = array();

foreach ($params as $paramKey => $paramValue) {

    if ($isNumeric || !$isArray) {
        $finalParams[$paramKey] = $paramValue;
        continue;
    }

    $placeholders = $this->placeholders($paramKey, $paramValue);
    $finalParams += $placeholders
}

$params = $finalParams;
```

Reply:
right, it's safer to use a temp var. tks.